ui/src/app/views/resource-card.tsx

```diff
@@ -439,51 +433,47 @@ export class ResourceCard extends React.Component<Props, State> {
     }
   }
 
-  openResource(jupyterLab?: boolean): void {
+  getResourceUrl(jupyterLab?: boolean): string {
+    const {workspaceNamespace, workspaceFirecloudName, conceptSet, notebook, dataSet, cohort} =
+      this.props.resourceCard;
+    const workspacePrefix = `/workspaces/${workspaceNamespace}/${workspaceFirecloudName}`;
+
     switch (this.resourceType) {
       case ResourceType.COHORT: {
-        this.reviewCohort();
-        break;
+        return `${workspacePrefix}/cohorts/${cohort.id}/review`;
       }
       case ResourceType.CONCEPT_SET: {
-        navigate(['workspaces', this.props.resourceCard.workspaceNamespace,
-          this.props.resourceCard.workspaceFirecloudName, 'concepts', 'sets',
-          this.props.resourceCard.conceptSet.id], {relativeTo: null});
-        break;
+        return `${workspacePrefix}/concepts/sets/${conceptSet.id}`;
       }
       case ResourceType.NOTEBOOK: {
         const queryParams = {
           playgroundMode: false,
-          jupyterLabMode: jupyterLab
+          jupyterLabMode: jupyterLab,
         };
         if (this.notebookReadOnly) {
           queryParams.playgroundMode = true;
         }
-        navigate(
-          ['workspaces', this.props.resourceCard.workspaceNamespace,
-            this.props.resourceCard.workspaceFirecloudName, 'notebooks',
-            encodeURIComponent(this.props.resourceCard.notebook.name)], {
-              queryParams,
-              relativeTo: null,
-            });
-        break;
+        return `${workspacePrefix}/notebooks/${encodeURIComponent(notebook.name)}?` +
+          fp.toPairs(queryParams).map(kvp => `${kvp[0]}=${kvp[1]}`).join('&');
```

Comment:
Optional: there should be a simpler & safer way to render a set of query params to string... maybe use the URL API (https://developer.mozilla.org/en-US/docs/Web/API/URL) with

new URLSearchParams(queryParams).toString()